src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java

```diff
@@ -121,6 +144,204 @@ public void importRepositories(final ProgrammingExercise templateExercise, final
+            case JAVA, KOTLIN -> {
+
+                copyExerciseContentToRepositoryForJavaOrKotlinExercise(templateRepo, RepositoryType.TEMPLATE, basePath, newExercise, oldPackageName);
+                copyExerciseContentToRepositoryForJavaOrKotlinExercise(solutionRepo, RepositoryType.SOLUTION, basePath, newExercise, oldPackageName);
+                copyExerciseContentToRepositoryForJavaOrKotlinExercise(testRepo, RepositoryType.TESTS, basePath, newExercise, oldPackageName);
+            }
+            case C, VHDL, ASSEMBLER -> {
+                copyExerciseContentToRepositoryForExerciseWithPlainDirectoryStructure(templateRepo, RepositoryType.TEMPLATE, basePath);
+                copyExerciseContentToRepositoryForExerciseWithPlainDirectoryStructure(solutionRepo, RepositoryType.SOLUTION, basePath);
+                copyExerciseContentToRepositoryForExerciseWithPlainDirectoryStructure(testRepo, RepositoryType.TESTS, basePath);
+            }
+            case PYTHON -> {
+                copyExerciseContentToRepositoryForPythonExercise(templateRepo, RepositoryType.TEMPLATE, basePath);
+                copyExerciseContentToRepositoryForPythonExercise(solutionRepo, RepositoryType.SOLUTION, basePath);
+                copyExerciseContentToRepositoryForPythonExercise(testRepo, RepositoryType.TESTS, basePath);
+            }
+            case OCAML -> {
+                copyExerciseContentToRepositoryForOcamlExercise(templateRepo, RepositoryType.TEMPLATE, basePath);
+                copyExerciseContentToRepositoryForOcamlExercise(solutionRepo, RepositoryType.SOLUTION, basePath);
+                copyExerciseContentToRepositoryForOcamlExercise(testRepo, RepositoryType.TESTS, basePath);
+            }
+            case HASKELL -> {
+                copyExerciseContentToRepositoryForHaskellExercise(templateRepo, RepositoryType.TEMPLATE, basePath);
+                copyExerciseContentToRepositoryForHaskellExercise(solutionRepo, RepositoryType.SOLUTION, basePath);
+                copyExerciseContentToRepositoryForHaskellExercise(testRepo, RepositoryType.TESTS, basePath);
+            }
+        }
+    }
+
+    private void copyExerciseContentToRepositoryForExerciseWithPlainDirectoryStructure(Repository repository, RepositoryType repositoryType, Path exerciseDir) throws IOException {
+        for (Path file : retrieveRepositoryContentPaths(retrieveRepositoryDirectoryPath(exerciseDir, repositoryType.getName()))) {
+            repositoryService.createFile(repository, String.valueOf(file.getFileName()), Files.newInputStream(file));
+
+        }
+
+    }
+
+    private void copyExerciseContentToRepositoryForPythonExercise(Repository repository, RepositoryType repositoryType, Path exerciseDir) throws IOException {
+        boolean testRepo;
+        if (repositoryType == RepositoryType.TESTS) {
+            repositoryService.createFolderRecursively(repository, "behavior");
+            repositoryService.createFolderRecursively(repository, "structural");
```

Comment:
Some of the names like `behavior`, `structural`, `src` are reused quite a few times. Introducing appropriate constants would be nice.